File: montecarlo/python/box_muller.py
import numpy as np
import matplotlib.pyplot as plt
import math as m

gen = np.random.RandomState()
gen.seed(123456789)

N_points = 10000
n_bins  = 30

values = np.array([])

for i in range(0,N_points):
    eta_1 = gen.random()
    eta_2 = gen.random()

    x = m.sqrt(-2*m.log(eta_1))*m.cos(2*m.pi*eta_2)
    values = np.append(values,x)

plt.hist(values, n_bins)
plt.show()

File: montecarlo/cpp/box_muller.cpp
#include <iostream>
#include <TApplication.h>
#include <TMath.h>
#include <TRandom3.h>
#include <TH1D.h>
#include <cmath>

using namespace std;

int main(){

    TApplication app("app",NULL,NULL);

    //generator
    TRandom3 rnd;
    rnd.SetSeed(123456789);

    //parameters
    int N_points = 1000000;
    int n_bins  = 40;
    float eta_1;
    float eta_2;
    float x;

    //histogram
    TH1D *h = new TH1D("h","Gaussian distribution",n_bins,-4,4);

    for (int i=0; i<N_points; i++){
        eta_1 = rnd.Rndm();
        eta_2 = rnd.Rndm();

        x=sqrt(-2*TMath::Log(eta_1))*TMath::Cos(2*M_PI*eta_2);
        h->Fill(x);
    }

    h->Draw();
    app.Run(true);
    return 0;
}